toml-test/meson.build

```diff
@@ -1,9 +1,14 @@
+tt_deps = [
+	tomlplusplus_dep,
+	dependency('nlohmann_json', fallback: ['json', 'nlohmann_json_dep'])
```

Comment:
This does not require the same `declare_dependency()` hacks that I used in tests/meson.build because the `json` submodule contains a `meson.build` file